website/addons/dataverse/views/config.py

```diff
@@ -21,7 +21,7 @@ def dataverse_get_user_accounts(auth):
     """ Returns the list of all of the current user's authorized Dataverse accounts """
 
     return DataverseSerializer(
-        user_settings=auth.user.get_addon('dataverse')
+        user_settings=auth.user.get_or_add_addon('dataverse')
```

Comment:
This line adds the addon to the user if it doesn't exist. That's the right thing to do here if the user has an `ExternalAccount` for Dataverse, because that can't happen without the user going through the OAuth flow.

It's not the right thing to do if the user hasn't gone through the OAuth flow though - otherwise, the act of going to the node settings page for a project with Dataverse enabled will enable the Dataverse addon for my user, regardless of whether or not I have a Dataverse account associated with my user.

Let's put a check in there first, and only add the user settings to the user if that user has a Dataverse `ExternalAccount` attached to their user.

Alternatively, we could create the user addon during the "callback", and modify `DataverseSerializer` to handle the case where the user addon was `None`.